### ISSUE TITLE
Ability to create projects from scratch

### ISSUE THREAD

**Issue by jimon** [kronenthaler/mod-pbxproj#87]
I know that this project mainly meant to be used in read-modify-write fashion. Is there anything in particular stopping us from actually creating projects from scratch?

When I just trying to create XcodeProject instance directly by doing ```prj = XcodeProject()``` I get crash on ```for k, v in self.objects.iteritems():``` [here](https://github.com/kronenthaler/mod-pbxproj/blob/master/mod_pbxproj/mod_pbxproj.py#L608) with objects being NoneType. It's probably just a minor thing :) Should I expect anything major?

PS. This sort of feature is useful for project generators in build systems :)

**Comment by kronenthaler** [comment]
True, this project started as something to modify Unity's generated projects. I never really thought of about creating new projects from scratch. What i have seen before is the use of a seed project. A very simple project that has some placeholders here and there that are replaced by a tool to generate your initial project. Maybe that is the alternative that can work for you at the moment.

I will mark this ticket as an enhancement, and will pick it up in the future.

**Comment by garvankeeley** [comment]
I have used the seed project method successfully, the only placeholder I had was an empty target, with all settings defined in xcconfig files, so the pbxproj file ends up quite small and minimal. All files and additional build steps populated using mod-pbxproj.

**Comment by Diarrhio** [comment]
Would love to see this too. We've hand rolled our own Xcode project writing system in our build system (starting many years ago), but would love to switch to this library. Only thing preventing us is that is the lack of this feature, which admittedly seems like a big feature to implement

**Comment by paulo-coutinho** [comment]
Hi,

Anyone have a demo about how to make it from scratch?

Thanks.
